Assets/VRTK/Scripts/Locomotion/VRTK_SlingshotJump.cs

```diff
@@ -0,0 +1,239 @@
+            if (!rightIsAiming && rightControllerEvents.IsButtonPressed(triggerButton) && !IsClimbing())
+            {
+                rightIsAiming = true;
+                rightStartAimPosition = playArea.InverseTransformPoint(rightControllerEvents.gameObject.transform.position);
+            }
+
+            // Check for release states
+            if (leftIsAiming && !leftControllerEvents.IsButtonPressed(triggerButton))
+            {
+                leftReleasePosition = playArea.InverseTransformPoint(leftControllerEvents.gameObject.transform.position);
+                if (!rightTriggerReleased)
+                {
+                    countDownEndTime = Time.time + releaseWindowTime;
+                }
+                leftTriggerReleased = true;
+            }
+
+            if (rightIsAiming && !rightControllerEvents.IsButtonPressed(triggerButton))
+            {
+                rightReleasePosition = playArea.InverseTransformPoint(rightControllerEvents.gameObject.transform.position);
+                if (!leftTriggerReleased)
+                {
+                    countDownEndTime = Time.time + releaseWindowTime;
+                }
+                rightTriggerReleased = true;
+            }
+
+            // Check for reset
+            if ((leftTriggerReleased || rightTriggerReleased) && Time.time > countDownEndTime)
+            {
+                UnAim();
+            }
+
+            // Check for jump
+            if (leftTriggerReleased && rightTriggerReleased && !bodyPhysics.IsFalling())
+            {
+                Vector3 leftDir = leftStartAimPosition-leftReleasePosition;
+                Vector3 rightDir = rightStartAimPosition-rightReleasePosition;
+                Vector3 localJumpDir = leftDir + rightDir;
+                Vector3 worldJumpDir = playArea.transform.TransformVector(localJumpDir);
+                Vector3 jumpVector = worldJumpDir * velocityMultiplier;
+                
+                if (jumpVector.magnitude > velocityMax)
+                {
+                    jumpVector = jumpVector.normalized * velocityMax;
+                } 
+
+                bodyPhysics.ApplyBodyVelocity(jumpVector, true, true);
+
+                UnAim();
+
+                OnSlingshotJumped(SetPlayerClimbEvent(gameObject));
+            }
+        }
+
+        protected void OnSlingshotJumped(SlingshotJumpEventArgs e)
+        {
+            if (SlingshotJumped != null)
+            {
+                SlingshotJumped(this, e);
+            }
+        }
+
+        protected SlingshotJumpEventArgs SetPlayerClimbEvent(GameObject target)
+        {
+            SlingshotJumpEventArgs e;
+            e.target = target;
+            return e;
+        }
+
+        protected void InitListeners(bool state)
+        {
+            InitTeleportListener(state);
+        }
+
+        protected void InitTeleportListener(bool state)
+        {
+            var teleportComponent = GetComponent<VRTK_BasicTeleport>();
```

Comment:
best to just avoid `var` even though this is the correct usage, we're aiming to be more explicit so:

`VRTK_BasicTeleport teleportComponent = GetComponent<VRTK_BasicTeleport>();`